google/cloud/security/scanner/audit/forwarding_rule_rules_engine.py

```diff
@@ -0,0 +1,249 @@
+                resource_type=ResourceType.FORWARDING_RULE)
+
+    def add_rules(self, rules):
+        """Add rules to the rule book.
+
+        Args:
+            rules (dict): rule from file to be added to book
+        """
+        if self.rule_book is not None:
+            self.rule_book.add_rules(rules)
+
+
+class ForwardingRuleRulesBook(bre.BaseRuleBook):
+    """The RuleBook for forwarding rules resources."""
+
+    def __init__(self, rule_defs=None):
+        """Initialization.
+
+        Args:
+            rule_defs (dict): rule definitions
+        """
+        super(ForwardingRuleRulesBook, self).__init__()
+        self.resource_rules_map = {}
+        if not rule_defs:
+            self.rule_defs = {}
+        else:
+            self.rule_defs = rule_defs
+            self.add_rules(rule_defs)
+
+    def add_rules(self, rule_defs):
+        """Add rules to the rule book
+
+        Args:
+            rule_defs (dict): list of rules and their index number
+        """
+        for (i, rule) in enumerate(rule_defs.get('rules', [])):
+            self.add_rule(rule, i)
+
+    def add_rule(self, rule_def, rule_index):
+        """Add a rule to the rule book.
+
+        Args:
+            rule_def (dict): A dictionary containing rule definition
+                properties.
+            rule_index (int): The index of the rule from the rule definitions.
+            Assigned automatically when the rule book is built.
+
+        Raises:
+            InvalidRulesSchemaError: if rule has format error
+        """
+
+        target = rule_def.get('target')
+        mode = rule_def.get('mode')
+        load_balancing_scheme = rule_def.get('load_balancing_scheme')
+        port_range = rule_def.get('port_range')
+        port = rule_def.get('port')
+        ip_address = rule_def.get('ip_address')
+        ip_protocol = rule_def.get('ip_protocol')
+        if (target is None) or (mode is None) or \
+           (load_balancing_scheme is None) or \
+           (ip_address is None) or (ip_protocol is None):
+            raise audit_errors.InvalidRulesSchemaError(
+                'Faulty rule {}'.format(rule_def.get('name')))
+        rule_def_resource = {'target': target,
+                             'mode': mode,
+                             'load_balancing_scheme': load_balancing_scheme,
+                             'port_range': port_range,
+                             'ip_address': ip_address,
+                             'ip_protocol': ip_protocol,
+                             'port': port, }
+
+        rule = Rule(rule_name=rule_def.get('name'),
+                    rule_index=rule_index,
+                    rules=rule_def_resource)
+
+        resource_rules = self.resource_rules_map.get(rule_index)
+        if not resource_rules:
+            self.resource_rules_map[rule_index] = rule
+
+    def get_resource_rules(self):
+        """Get all the resource rules for (resource, RuleAppliesTo.*).
```

Comment:
looking at this method, it looks like the method description might be, "Get all the resource_rules as a list from the resource_rules_map". Would it be more concise to do:

```
return list(self.resource_rules_map.values())
```